mesh_segmenter/src/mesh_segmenter.cpp

```diff
@@ -33,28 +31,34 @@ std::vector<vtkSmartPointer<vtkPolyData> > MeshSegmenter::getMeshSegments()
   vtkSmartPointer<vtkPolyData> input_copy = vtkSmartPointer<vtkPolyData>::New();
   input_copy->DeepCopy(input_mesh_);
   std::vector<vtkSmartPointer<vtkPolyData> > meshes;
-  for(int i = 0; i < included_indices_.size(); ++i)
+  //TODO: Figure out why the last one fails for some meshes (and then remove the -1)
+  for (int i = 0; i < included_indices_.size() - 1; ++i)
   {
-
     vtkSmartPointer<vtkPolyData> mesh = vtkSmartPointer<vtkPolyData>::New();
+    // Create new pointer to a new copy of input_mesh_
     mesh->DeepCopy(input_mesh_);
+    // Initilize pointers to points/polys - Also resets them
     mesh->GetPoints()->Initialize();
     mesh->GetPolys()->Initialize();
     input_copy->GetCellData()->CopyNormalsOn();
 
     // Preallocate memory, this is NECESSARY or normal data is NOT copied
-    mesh->GetCellData()->CopyAllocate(input_copy->GetCellData(), included_indices_[i]->GetNumberOfIds(), included_indices_[i]->GetNumberOfIds());
-    mesh->GetPointData()->CopyAllocate(input_copy->GetPointData(), included_indices_[i]->GetNumberOfIds()*2, included_indices_[i]->GetNumberOfIds()*2);
+    mesh->GetCellData()->CopyAllocate(
+        input_copy->GetCellData(), included_indices_[i]->GetNumberOfIds(), included_indices_[i]->GetNumberOfIds());
+    mesh->GetPointData()->CopyAllocate(input_copy->GetPointData(),
+                                       included_indices_[i]->GetNumberOfIds() * 2,
+                                       included_indices_[i]->GetNumberOfIds() * 2);
 
     // copy cell data and normals
-    mesh->CopyCells(input_copy, included_indices_[i]);
+      mesh->CopyCells(input_copy, included_indices_[i]);
 
-    if(mesh->GetNumberOfCells() <= 1)
-    {
-      cout << "NOT ENOUGH CELLS FOR SEGMENTATION\n";
-      continue;
-    }
-    meshes.push_back(mesh);
+
+      if (mesh->GetNumberOfCells() <= 1)
+      {
+        cout << "NOT ENOUGH CELLS FOR SEGMENTATION\n";
```

Comment:
cout instead of ROS_INFO or ROS_ERROR or ROS_ERROR_STREAM?

Reply:
At the moment, the way noether is structured is that this segmentation package as well as I believe the sequence planner and tool path planner are standalone libraries. Then noether adds those libraries and makes it ROS specific. Thus, I did not want to use ROS_INFO because this package is currently independent of ROS.